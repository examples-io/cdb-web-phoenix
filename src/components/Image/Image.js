import React, { Component } from 'react'
import PropTypes from 'prop-types'

/**
 * @description Image component - renders an accessible image
 */
class Image extends Component {
    render() {
        return (
            <img className={this.props.className}
                 id={this.props.id}
                 src={this.props.src}
                 alt={this.props.alt} />
        );
    }
}

Image.propTypes = {
    /**
     * className - classes to be applied to the image element
     */
    className: PropTypes.string,
    /**
     * id - element id of the image
     */
    id: PropTypes.string,
    /**
     * src - location of the image
     */
    src: PropTypes.node,
    /**
     * alt - alternative text for the image (for accessibility purposes)
     */
    alt: PropTypes.string
};

export default Image;