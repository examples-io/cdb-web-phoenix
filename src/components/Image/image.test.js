import React from 'react';
import renderer from 'react-test-renderer';

import Image from './Image';

describe('Image tests', () => {
    it ('renders Image correctly', () => {
        const tree = renderer
            .create(<Image id="scotiabankLogo" alt="Scotiabank logo" src="" />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});