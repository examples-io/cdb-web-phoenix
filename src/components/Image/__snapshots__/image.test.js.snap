// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image tests renders Image correctly 1`] = `
<img
  alt="Scotiabank logo"
  className={undefined}
  id="scotiabankLogo"
  src=""
/>
`;
